drakrun/drakrun/storage.py

```diff
@@ -188,18 +197,38 @@ def export_vm0(self, file):
             )
 
     def import_vm0(self, file):
-        # Clean ZFS dataset
-        subprocess.run(
-            ["zfs", "destroy", "-r", "self.zfs_tank_name"],
-            check=True
-        )
+        self.delete_zfs_tank()
         with open(file, "rb") as snapshot_file:
             subprocess.run(
                 ["zfs", "recv", f"{self.zfs_tank_name}/vm-0@booted"],
                 check=True,
                 stdout=snapshot_file
             )
 
+    def delete_vm_volume(self, vm_id: int) -> bool:
+        vm_id_vol = shlex.quote(os.path.join(self.zfs_tank_name, f"vm-{vm_id}"))
+        try:
+            logging.info(f"Deleting zfs volume {vm_id_vol}")
+            subprocess.check_output(
+                f"zfs destroy -Rfr {vm_id_vol}", stderr=subprocess.STDOUT, shell=True
+            )
```

Comment:
Instead of quoting, we can just avoid using `shell=True`

Reply:
is `shell=True` insecure or something?

Reply:
IMO it's just ugly. You're forced to quote and perform string interpolation instead of calling subprocess with a argument list

Reply:
😆 😆 fine, will change this